src/NosCore.Parser/Parsers/ItemParser.cs

```diff
@@ -0,0 +1,679 @@
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using ChickenAPI.Packets.Enumerations;
+using NosCore.Core;
+using NosCore.Core.I18N;
+using NosCore.Data;
+using NosCore.Data.Enumerations.I18N;
+using NosCore.Data.Enumerations.Items;
+using NosCore.Data.StaticEntities;
+using NosCore.Parser.Parsers.Generic;
+using Serilog;
+
+namespace NosCore.Parser.Parsers
+{
+    public class ItemParser
+    {
+        //  VNUM	{VNum}	{Price}
+        //	NAME {Name}
+
+        //  INDEX	0	0	0	0	0	0
+        //	TYPE	0	0
+        //	FLAG	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
+        //	DATA	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
+        //	BUFF	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
+        //	LINEDESC	0
+        //{Desc}
+        //    END
+        //#========================================================
+        private const string ItemCardDto = "\\Item.dat";
+
+        private readonly IGenericDao<ItemDto> _itemDao;
+        private readonly IGenericDao<BCardDto> _bcardDao;
+        private readonly ILogger _logger;
+
+        public ItemParser(IGenericDao<ItemDto> itemDao, IGenericDao<BCardDto> bCardDao, ILogger logger)
+        {
+            _itemDao = itemDao;
+            _bcardDao = bCardDao;
+            _logger = logger;
+        }
+
+        public void Parse(string folder)
+        {
+            var actionList = new Dictionary<string, Func<Dictionary<string, string[][]>, object>>
+            {
+                {nameof(ItemDto.VNum), chunk => Convert.ToInt16(chunk["VNUM"][0][2])},
+                {nameof(ItemDto.Price), chunk => Convert.ToInt64(chunk["VNUM"][0][3])},
+                {nameof(ItemDto.ReputPrice), chunk => chunk["FLAG"][0][21] == "1" ? Convert.ToInt64(chunk["VNUM"][0][3]) : 0},
+                {nameof(ItemDto.NameI18NKey), chunk => chunk["NAME"][0][2]},
+                {nameof(ItemDto.Type), chunk => ImportType(chunk)},
+                {nameof(ItemDto.ItemType), chunk => ImportItemType(chunk)},
+                {nameof(ItemDto.ItemSubType), chunk => Convert.ToByte(chunk["INDEX"][0][4])},
+                {nameof(ItemDto.EquipmentSlot), chunk => ImportEquipmentType(chunk)},
+                {nameof(ItemDto.Morph), chunk =>  ImportEffect(chunk) == ItemEffectType.ApplySkinPartner ?Convert.ToInt16(chunk["INDEX"][0][5]) :
+                    ImportEquipmentType(chunk) != EquipmentType.Amulet ? Convert.ToInt16(chunk["INDEX"][0][7]) : default},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Move this closing curly brace to the next line.](https://app.codacy.com/app/NosCoreIO/NosCore/pullRequest?prid=4456396)